tsnet: flaky test in TestUserMetricsRouteGauges

https://github.com/tailscale/tailscale/actions/runs/18662546634/job/53206259954

```
=== RUN   TestUserMetricsRouteGauges
    tsnet_test.go:117: DERP httpsrv listener: 127.0.0.1:42283
    tsnet_test.go:130: testcontrol listening on http://127.0.0.1:38147/
2025/10/20 19:26:36 tsnet starting with hostname "s1", varRoot "/tmp/TestUserMetricsRouteGauges443641201/001/s1"
2025/10/20 19:26:36 LocalBackend state is NeedsLogin; running StartLoginInteractive...
    testcontrol.go:315: Created user userid:1 for node nodekey:8fd627769ca7d18baf88df150785814c417810748afeaac4637bfdae9f985a6a
2025/10/20 19:26:36 tsnet starting with hostname "s2", varRoot "/tmp/TestUserMetricsRouteGauges443641201/002/s2"
2025/10/20 19:26:36 LocalBackend state is NeedsLogin; running StartLoginInteractive...
    testcontrol.go:315: Created user userid:2 for node nodekey:d66039cacc789a8989710f2f9842c3d40e3bd6c52e64922ad9c9bcd79b971805
    testcontrol.go:315: Setting subnet routes for [j9Ynd]: [192.0.2.0/24 192.0.5.1/32 0.0.0.0/0]
2025/10/20 19:26:41 AuthLoop: state is Running; done
2025/10/20 19:26:41 AuthLoop: state is Running; done
    tsnet_test.go:1202: waiting for condition: primary routes available for node1
    stuntest.go:65: STUN server shutdown
--- FAIL: TestUserMetricsRouteGauges (91.30s)
```

Similar:
https://github.com/tailscale/tailscale/actions/runs/18785428700/job/53602366507
https://github.com/tailscale/tailscale/actions/runs/19439824863/job/55620111391

--- a/tstest/integration/testcontrol/testcontrol.go
+++ b/tstest/integration/testcontrol/testcontrol.go
@@ -464,6 +464,9 @@ func (s *Server) SetSubnetRoutes(nodeKey key.NodePublic, routes []netip.Prefix) 
 	defer s.mu.Unlock()
 	s.logf("Setting subnet routes for %s: %v", nodeKey.ShortString(), routes)
 	mak.Set(&s.nodeSubnetRoutes, nodeKey, routes)
+	if node, ok := s.nodes[nodeKey]; ok {
+		sendUpdate(s.updates[node.ID], updateSelfChanged)
+	}
 }
 
 // MasqueradePair is a pair of nodes and the IP address that the

